From 8d4c6ec6352c3a828c7ba812d87dd4247fbc840c Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Sat, 29 Jan 2022 02:10:37 +0200
Subject: [PATCH] [SW-65638] gaudi2: release: WIP remote fwl

Operate the host's hl-fw-loader locally.

TODO: add a flag to load local file as itb and run it remotely

Change-Id: Ibe663dc18db2f2123b7c9131131e7083c42675a6
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/scripts/fwi/fw_installer.sh b/scripts/fwi/fw_installer.sh
index 8f61f3c..317bb89 100755
--- a/scripts/fwi/fw_installer.sh
+++ b/scripts/fwi/fw_installer.sh
@@ -259,6 +259,7 @@
 	--skip-install            : skip the tools installation, can be used on an already visited system
 	--skip-copy               : skip the files copy, can be used on an already visited system, working
 
+	--fwl [argumenst]         : execute the hl-fw-loader remotly from the local host. all arguments passed the remote applcation
 
         Mandatory
 	=========
@@ -652,6 +653,18 @@
 			pok "[${pci_dev}] Succeed to update eeprom: $file"
 		}
 	done
+	return 0
+}
+
+function remote_fwl()
+{
+	$dry_run host_run sudo hl-fw-loader *$
+	[ $? -ne 0 ] && {
+		perr "[${pci_dev}] FWL failed"
+		return 1
+	} || {
+		pok "[${pci_dev}] FWL Succeed"
+	}
 	return 0
 }
 
@@ -1083,6 +1096,14 @@
 	return 0
 }
 
+function remote_fwl_op()
+{
+	[ ${fwl_f:-0} -eq 0 ] && return 0
+
+	#redirect remote_fwl ${fwl_args[@]}
+	remote_fwl ${fwl_args[@]} >&3
+}
+
 function handle_ignore_op()
 {
 	[ ${#ignore_args[@]} -eq 0 ] && ignore_args=(host)
@@ -1240,6 +1261,11 @@
 			nargs=$#
 			# not using shift cause  the number of args was already calculated
 			unset rmt_run_args[0]
+			;;
+		--fwl)
+			fwl_f=1
+			fwl_args=($*)
+			#unset fwl_args[0]
 			;;
 		--console|--oam-con-out)
 			rmt_run=bmc
@@ -1902,6 +1928,8 @@
 	mods_var_register create_cfg snapshot
 	mods_var_register config_link show_env
 	mods_var_register copy_files
+	mods_var_register fwl
+
 	# User config files and platform selection was already
 	# handled in the early param handling
 	load_platform_api ${platform_type}
